control-plane/pkg/reconciler/base/reconciler.go

```diff
@@ -243,6 +231,15 @@ func (r *Reconciler) updatePodsAnnotation(ctx context.Context, logger *zap.Logge
 			annotations = make(map[string]string, 1)
 		}
 
+		// Check whether pod's annotation is the expected one.
+		if v, ok := annotations[VolumeGenerationAnnotationKey]; ok {
+			v, err := strconv.ParseUint(v /* base */, 10 /* bitSize */, 64)
+			if err == nil && v == volumeGeneration {
```

Comment:
maybe using `v >= volumeGeneration` it's even safer, since this number is incremental?

Reply:
We can start from 0 again and so `v >= volumeGeneration` is not ok:

https://github.com/knative-sandbox/eventing-kafka-broker/blob/bb90971f62275a019930278f2e72c5361921461d/control-plane/pkg/core/config/utils.go#L125

Reply:
It might happen since we're using the cache but we will retry later on and it will eventually converge, so as long as our CM is consistent we are good.

Reply:
That might happen even today without this PR.